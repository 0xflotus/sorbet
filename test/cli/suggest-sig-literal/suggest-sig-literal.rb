# typed: strict
def index_for_live(fields)
  [[:deleted_at, 1]] + fields
end
