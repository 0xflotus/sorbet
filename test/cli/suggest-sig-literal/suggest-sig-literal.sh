#!/bin/bash

tmp="$(mktemp)"
infile="test/cli/suggest-sig-literal/suggest-sig-literal.rb"
cp "$infile" "$tmp"

cwd="$(pwd)"
cd "$(dirname "$tmp")" || exit 1
cp "$tmp" suggest-sig-literal.rb

"$cwd/main/sorbet" --silence-dev-message -a suggest-sig-literal.rb 2>&1

echo
echo --------------------------------------------------------------------------
echo

# Also cat the file, to make sure that 'extend' is only added once per class.
cat suggest-sig-literal.rb

rm suggest-sig-literal.rb
rm "$tmp"
