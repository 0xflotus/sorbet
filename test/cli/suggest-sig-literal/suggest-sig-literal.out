suggest-sig-literal.rb:2: This function does not have a `sig` http://go/e/7017
     2 |def index_for_live(fields)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
  Autocorrect: Done
    suggest-sig-literal.rb:2: Replaced with `sig {generated.params(fields: T::Enumerable[T::Array[T.any(Symbol, Integer)]]).returns(T::Array[T::Array[T.any(Symbol, Integer)]])}
`
     2 |def index_for_live(fields)
        ^
Errors: 1

--------------------------------------------------------------------------

# typed: strict
sig {generated.params(fields: T::Enumerable[T::Array[T.any(Symbol, Integer)]]).returns(T::Array[T::Array[T.any(Symbol, Integer)]])}
def index_for_live(fields)
  [[:deleted_at, 1]] + fields
end
